x/posts/internal/types/post_media.go

```diff
@@ -0,0 +1,120 @@
+package types
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// ---------------
+// --- PostMedia
+// ---------------
+
+type PostMedia struct {
+	URI      string `json:"uri"`
+	MimeType string `json:"mime_Type"`
+}
+
+func NewPostMedia(uri, mimeType string) PostMedia {
+	return PostMedia{
+		URI:      uri,
+		MimeType: mimeType,
+	}
+}
+
+// String implements fmt.Stringer
+func (pm PostMedia) String() string {
+	out := "Media - "
+
+	out += fmt.Sprintf(" URI - [%s] ; Mime-Type - [%s] \n", pm.URI, pm.MimeType)
+
+	return strings.TrimSpace(out)
```

Comment:
I would advise not splitting this into three different like, but doing all into one line 

```golang
return fmt.Sprintf("%s (%s)", pm.URI, pm.MimeType)
```